internal/pkg/gateway/api.go

```diff
@@ -134,26 +134,42 @@ func (gs *Server) Endorse(ctx context.Context, request *gp.EndorseRequest) (*gp.
 	}
 	proposal, err := protoutil.UnmarshalProposal(signedProposal.ProposalBytes)
 	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "failed to unpack transaction proposal: %s", err)
+		return nil, status.Error(codes.InvalidArgument, err.Error())
```

Comment:
Unrelated to this PR, but... this Endorse function is over 200 lines long. 

Shouldn't it be shorter and easier to parse? you have down the road a goroutine which spawns another goroutine inside a loop, this is very brittle and easy to introduce bugs with time.

Reply:
Yes, good point.  I've raised a separate issue (https://github.com/hyperledger/fabric/issues/3122) to track this.